config/initializers/rack_middleware.rb

```diff
@@ -11,7 +11,8 @@
   optional: [
     /^\/authn-[^\/]+\//,
     /^\/authn\//,
-    /^\/public_keys\//
+    /^\/public_keys\//,
+    /^\/whoami$/
```

Comment:
If we make this token required for `whoami` can it work for you? related to comment above

Reply:
Sure, we can make this require authentication. I'm not sure I agree with the cost/risk of echoing the client IP address, but for my use of this, I can use authenticated requests.

CC: @alexkalish @andytinkham if either of you have comments/thoughts.

Reply:
I'm less concerned about echoing the client IP back (my understanding is that it would basically be the equivalent of https://whatsmyip.com/your-ip-address, right?). I think having it require authentication addresses my concern about increasing the impact of an API token leak.

Reply:
@andytinkham 

> I'm less concerned about echoing the client IP back (my understanding is that it would basically be the equivalent of https://whatsmyip.com/your-ip-address, right?)

Similar, yes. However, assuming this in an internal network, if the XFF is not configured correctly, it could potentially reveal the address of a load balancer, for example.

> I think having it require authentication addresses my concern about increasing the impact of an API token leak.

Ironically, this doesn't do anything to address the token info concern. If you have a leaked access token that is still valid, you can use it to call this endpoint. But this doesn't reveal any information not contained in the token in plain-text directly.